packages/@o3r/rules-engine/schematics/rules-engine-to-component/index.ts

```diff
@@ -0,0 +1,138 @@
+import { chain, noop, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import {
+  applyEsLintFix,
+  findMethodByName,
+  fixStringLiterals,
+  generateBlockStatementsFromString,
+  generateClassElementsFromString,
+  generateImplementsExpressionWithTypeArguments,
+  getLibraryNameFromPath,
+  getSimpleUpdatedMethod, sortClassElement
+} from '@o3r/schematics';
+import { insertImport } from '@schematics/angular/utility/ast-utils';
+import { applyToUpdateRecorder, Change } from '@schematics/angular/utility/change';
+import * as ts from 'typescript';
+import { NgGenerateRulesEngineToComponentSchematicsSchema } from './schema';
+
+/**
+ * Generate the code to enable rules-engine on a component
+ *
+ * @param options
+ */
+export function ngGenerateRulesEngineToComponent(options: NgGenerateRulesEngineToComponentSchematicsSchema): Rule {
+
+  const generateFiles: Rule = (tree: Tree, _context: SchematicContext) => {
+    if (options.path && tree.exists(options.path)) {
+      const projectName = options.projectName || getLibraryNameFromPath(options.path) || '';
+      const imports = [
+        {
+          from: '@angular/core',
+          importNames: ['inject', 'OnInit', 'OnDestroy']
+        },
+        {
+          from: '@o3r/configuration',
+          importNames: ['computeConfigurationName']
+        },
+        {
+          from: '@o3r/rules-engine',
+          importNames: ['RulesEngineService']
+        }
+      ];
+      let sourceFile = ts.createSourceFile(
+        options.path,
+        tree.readText(options.path),
+        ts.ScriptTarget.ES2020,
+        true
+      );
+      const recorder = tree.beginUpdate(options.path);
+      const changes = imports.reduce((acc: Change[], { importNames, from }) => acc.concat(
+        importNames.map((importName) =>
+          insertImport(sourceFile, options.path!, importName, from)
+        )
+      ), []);
+      applyToUpdateRecorder(recorder, changes);
+      tree.commitUpdate(recorder);
+
+      sourceFile = ts.createSourceFile(
+        options.path,
+        tree.readText(options.path),
+        ts.ScriptTarget.ES2020,
+        true
+      );
+      const result = ts.transform(sourceFile, [
+        (ctx) => (rootNode: ts.Node) => {
+          const { factory } = ctx;
+          const visit = (node: ts.Node): ts.Node => {
+            if (ts.isClassDeclaration(node)) {
+              const implementsClauses = node.heritageClauses?.find((heritageClause) => heritageClause.token === ts.SyntaxKind.ImplementsKeyword);
+              const interfaceToImplements = generateImplementsExpressionWithTypeArguments('OnInit, OnDestroy');
```

Comment:
you don't check the interface are not already in the file?

Reply:
except if the deduplicateHeritageCauses also works on interface?

Reply:
yes the naming is a bit misleading but `heritageClause` contains both extends and implements